packages/angular_devkit/build_angular/src/protractor/index.ts

```diff
@@ -44,7 +44,11 @@ export class ProtractorBuilder implements Builder<ProtractorBuilderOptions> {
     const options = builderConfig.options;
     const root = this.context.workspace.root;
     const projectRoot = resolve(root, builderConfig.root);
-    // const projectSystemRoot = getSystemPath(projectRoot);
+
+    // ensure that either one of this option is used
+    if (options.devServerTarget && options.baseUrl) {
+      throw new Error(`The 'baseUrl' option cannot be used with 'devServerTarget'.`);
```

Comment:
I think it makes sense to error out since we use the devServerTarget to compute baseUrl. But we should be explicit here. Something like:

```
The 'baseUrl' option cannot be used with 'devServerTarget'.
When present, 'devServerTarget' will be used to automatically setup 'baseUrl' for Protractor.
```

Reply:
I would actually expect it to override what's in the dev server target.  Otherwise, the baseUrl setting seems unnecessary as the overwhelming usage would be with a dev server target.

Reply:
What if host or port are also provided?

Reply:
The same.  Who knows how the developer has the environment setup. Or how they want protractor to connect to the application.

Reply:
I mean, if host and or port are provided with baseUrl which one should override which?

Reply:
Well, that's really a different question.  But I suppose it raises the additional question of why there are two ways to do the same thing?

Reply:
I’ll put this up for discussion.